src/hooks/persisted/useNovel.ts

```diff
@@ -253,16 +300,56 @@ export const useNovel = (novelPath: string, pluginId: string) => {
 
   const markPreviousChaptersUnread = (chapterId: number) => {
     if (novel) {
+      // Track progress if the plugin supports it
+      const plugin = getPlugin(novel.pluginId);
+      if (plugin?.trackProgress) {
+        const chapterPosition = chapters.find(
+          chapter => chapter.id === chapterId,
+        )?.position;
+        const trackedChapter =
+          chapters.find(chapter => chapter.position === chapterPosition! - 1) ||
+          chapters.find(chapter => chapter.position === chapterPosition);
```

Comment:
Why? This just feels wrong. Maybe add an explanation as this could easily lead to bugs.